# Q1. What are the three measures of central tendency?


There are three measures of central tendency :

Mean: The average of all the values in a data set.

Median: The middle value in a data set when it is ordered from least to greatest.

Mode: The most frequent value in a data set.

# Q2. What is the difference between the mean, median, and mode? How are they used to measure the central tendency of a dataset?


Mean is the average of all the values in a data set. It is calculated by adding all the values and dividing by the number of values.

Median is the middle value in a data set when it is ordered from least to greatest. If there is an even number of values, the median is the average of the two middlemost values.

Mode is the most frequent value in a data set.

Each measure of central tendency has its own strengths and weaknesses. The mean is the most commonly used measure of central tendency, but it can be misleading if there are outliers in the data set. The median is more resistant to outliers, but it does not take into account all the values in the data set. The mode is the least affected by outliers, but it can be misleading if there are multiple modes in the data set.

The best measure of central tendency to use depends on the specific data set and the purpose of the analysis.

Here are some examples of when to use each measure of central tendency:

Use the mean to get a general sense of the center of the data set.
Use the median if there are outliers in the data set.
Use the mode if you are interested in the most common value in the data set.

# Q3. Measure the three measures of central tendency for the given height data:

[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]


In [6]:
import numpy as np
from scipy import stats

data = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [5]:
# mean
mean = np.mean(data)

# median
median = np.median(data)

# mode
mode = stats.mode(data)

In [6]:
print( f"Mean: {mean}\nMedian: {median}\nMode: {mode}" )

Mean: 177.01875
Median: 177.0
Mode: ModeResult(mode=177.0, count=3)


# Q4. Find the standard deviation for the given data:

[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]


In [7]:
data = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

1st Method

In [8]:
var = np.var(data)
std = np.sqrt(var)

print(f"variance: {var}\nstandard deviation: {std}")

variance: 3.199023437500001
standard deviation: 1.7885814036548633


2nd Method

In [9]:
std = np.std(data)
print( f"Standard deviation: {std}")

Standard deviation: 1.7885814036548633


# Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe the spread of a dataset? Provide an example.


The range is the difference between the largest and smallest values in a dataset. It is a simple measure of dispersion that can be calculated quickly, but it can be influenced by outliers.

Variance is a more sophisticated measure of dispersion that takes into account how far each value is from the mean. It is calculated by taking the average of the squared differences between each value and the mean. A higher variance indicates that the values are more spread out from the mean.

Standard deviation is the square root of the variance. It is a commonly used measure of dispersion that is useful for interpreting the spread of data in relation to the mean. A higher standard deviation indicates that the values are more spread out from the mean.

In [22]:
#for example
data = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

range_ = max(data) - min(data)
variance = np.var(data)
std = np.std(data)

print(f"Range: {range_}\nVariance: {variance}\nStandard deviation: {std}")

Range: 7.5
Variance: 3.199023437500001
Standard deviation: 1.7885814036548633


# Q6. What is a Venn diagram?


A Venn diagram is a diagram that shows the logical relationships between a finite collection of different sets. Venn diagrams are typically used to compare and contrast two or more sets. The sets are represented by circles, and the overlaps between the circles represent the elements that are common to two or more sets.

For example, a Venn diagram could be used to show the relationships between the sets of students, athletes, and musicians. The circle for students would include all students, regardless of whether they are athletes or musicians. The circle for athletes would include all athletes, regardless of whether they are students or musicians. And the circle for musicians would include all musicians, regardless of whether they are students or athletes.

The overlaps between the circles would represent the students who are also athletes, the athletes who are also musicians, and the students who are also musicians.

<img src="VENN.gif" >

# Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:

(i) 	A n B


(ii)	A ⋃  B


In [18]:
A = {2,3,4,5,6,7}
B = {0,2,6,8,10}

In [19]:
intersection = A.intersection(B)
union = A.union(B)

In [20]:
print("A: ",A,"\nB: ",B,"\nIntersection: ",intersection, "\nUnion: ",union)

A:  {2, 3, 4, 5, 6, 7} 
B:  {0, 2, 6, 8, 10} 
Intersection:  {2, 6} 
Union:  {0, 2, 3, 4, 5, 6, 7, 8, 10}


# Q8. What do you understand about skewness in data?


Skewness is a measure of the asymmetry of a distribution. A distribution is asymmetrical when its left and right side are not mirror images. Skewness can be positive, negative, or zero.

_Positive skewness indicates that the tail of the distribution is longer on the right side, meaning that there are more data points with higher values than lower values. This is often seen in data sets that contain outliers, such as the distribution of income.

_Negative skewness indicates that the tail of the distribution is longer on the left side, meaning that there are more data points with lower values than higher values. This is often seen in data sets that are heavily skewed towards zero, such as the distribution of temperatures in a cold climate.

_Zero skewness indicates that the distribution is symmetrical, meaning that the left and right tails are of equal length. This is often seen in data sets that are normally distributed, such as the distribution of heights in a population.

<img src="sk1.png">

# Q9. If a data is right skewed then what will be the position of median with respect to mean?


If the data is right-skewed, there will be more data on the right side so the mean will surpass the median due to the longer tail on the right side of the distribution.

# Q10. Explain the difference between covariance and correlation. How are these measures used in statistical analysis?


Covariance and correlation are both measures of the relationship between two variables. However, they have different interpretations and uses.

_Covariance measures strength of the relationship between two variables. When two variables have a strong relationship, they tend to change together in a predictable way. It can have a positive, negative, or zero value, indicating whether the two variables are directly or inversely proportional.The covariance formula determines data points in a dataset from their average value. For instance, you can compute the Covariance between two random variables, X and Y, using the following formula:
<img src="cov2.png">
<img src="cov.png" >


_Correlation is a standardized version of covariance. It takes into account the scale of the two variables, so it is not affected by changes in units. The correlation can also be positive, negative, or zero.
A correlation of 1 indicates a perfect positive linear relationship between the two variables.
A correlation of -1 indicates a perfect negative linear relationship between the two variables.
A correlation of 0 indicates no linear relationship between the two variables.
Here is the formula :
<img src="cor.svg">



# Q11. What is the formula for calculating the sample mean? Provide an example calculation for a dataset.


In [1]:
data = [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

The formula for calculating the sample mean is:

x̄ = (x₁ + x₂ + ... + xn) / n

where x̄ is the sample mean, xi are the individual data points in the sample, and n is the sample size. For example:

In [2]:
mean = sum(data)/len(data)
print( f"The mean of data is {mean}" )

The mean of data is 177.01875


# Q12. For a normal distribution data what is the relationship between its measure of central tendency?


For data that follows a normal distribution, the three measures of central tendency (mean, median, and mode) are equal. This is because the normal distribution is symmetrical around its mean, and the mean, median, and mode all fall at the center of the distribution.

# Q13. How is covariance different from correlation?


1. Interpretation and Scale of Values:

_The change in scale changes the value of Covariance. A higher number in Covariance means higher dependency. Interpreting Covariance is difficult.
On the contrary, the correlation value remains unaffected by the change in scale. The correlation coefficients range from -1 to 1, which allows for a more straightforward interpretation, unlike Covariance.

2. Standardization and Comparison Across Datasets

_Since Covariance doesn’t have standardization, comparing covariances across various datasets is challenging. While correlation coefficients are standardized. Therefore, comparing it directly across variables, datasets, or contexts is easy. The range of covariance is from negative infinity to positive infinity, while correlation values lie between -1 and +1.

3. Robustness to Outliers

_Outlier hugely impacts the value of Covariance. Hence, it is sensitive to the presence of outliers. On the contrary, correlation coefficients offer a more robust standard of the relationship between variables, as correlation coefficients are less susceptible to outliers.

# Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.

Outliers can have a significant impact on measures of central tendency and dispersion.

For measures of central tendency, outliers can pull the mean away from the center of the distribution, making it an unreliable measure of the "typical" value. The median is less affected by outliers, as it is not influenced by extreme values. The mode may also be affected if the outlier is the most frequently occurring value in the dataset.

For measures of dispersion, outliers can greatly affect the range and standard deviation of the dataset. The range, which is the difference between the largest and smallest values in the dataset, can be greatly increased by a single outlier. The standard deviation, which measures the amount of variation in the dataset, can also be greatly affected by outliers.

For example, consider a dataset of test scores for a class of students:

In [12]:
data = [70,75,80,85,90,95,100,105]
print(f"Mean: {np.mean(data)}\nMedian: {np.median(data)}")

Mean: 87.5
Median: 87.5


If we add an outlier of 300 to the dataset:

In [13]:
data = [70,75,80,85,90,95,100,105,300]
print(f"Mean: {np.mean(data)}\nMedian: {np.median(data)}")

Mean: 111.11111111111111
Median: 90.0


In [ ]:
np

The mean score is now 111.111, which is significantly higher than the "typical" score for the class. The median score, on the other hand, is 90, which is a more accurate representation of the center of the distribution.